README.md

```diff
@@ -1,43 +1,466 @@
-  up-to-date than this README.
-- `copy --state <state name>`: Copy the given `state name` version of files to destination.
-  When Bluehawk encounters a `state` command (see below), multiple versions of the source file
-  are spawned. Each version removes any code in state commands that are **not**
-  marked with the corresponding state name. This flag determines which version
-  to eventually write to disk.
-- `snip`: Output snippet files only. Can be combined with `--state`.
-- `list commands`: Display a list of available markup commands.
-- `-d or --destination` defines the output location.
+`Main.codeblock.modulo.java.code-block.rst`:
+
+```rst
+.. code-block:: java
+   :emphasize-lines: 3
+
+   int dividend = 11;
+   int divisor = 3;
+   int modulus = dividend % divisor;
+   System.out.println(dividend + " % " + divisor + " = " + modulus);
+```
+
+### Remove
+
+The `remove` command, also aliased as `hide`, removes the spanned
+range from Bluehawk output. `remove` can be helpful for hiding
+assertions and state setup from user-facing code samples.
+You can use `remove` as either a block command or a
+line command.
+
+Consider the following file:
+
+`Main.java`:
+
+```java
+public class Main {
+
+  public static void main(String[] args){
+    // :code-block-start: division
+    int dividend = 11;
+    int divisor = 3;
+    int quotient = dividend / divisor;
+    assert(quotient == 3) // :remove:
+    System.out.println(dividend + " / " + divisor + " = " + quotient);
+    // :code-block-end:
+  }
+}
+// :replace-end:
+```
+
+Running the following command:
+
+```
+bluehawk snip Main.java -d .
+```
+
+Produces the following output:
+
+`Main.codeblock.division.java`:
+
+```rst
+int dividend = 11;
+int divisor = 3;
+int quotient = dividend / divisor;
+System.out.println(dividend + " / " + divisor + " = " + quotient);
+```
+
+## CLI
+
+### Commands
+
+Use commands to generate different kinds of output with Bluehawk, including
+code blocks, full files of code, and even error checks.
+
+> 💡 Commands for the Bluehawk CLI are not the same as
+> [Bluehawk Commands](#bluehawk-commands), the syntax
+> interpreted by Bluehawk to process input files.
+
+#### Snip
+
+```
+bluehawk snip --destination <output-directory> <input-directory-or-file>
+```
+
+Output "snippet files" that contain only the content of `code-block` or
+`snippet` Bluehawk commands, named in the format
+`<source-file-name>.codeblock.<codeblock-name>.<source-file-extension>`.
+By default, this command generates snippets
+that include only the _last_ (chronologically ordered in your file)
+state listed for each group of `state` Bluehawk commands. However,
+you can use the `--state` flag to generate snippet files that include
+content from a single state that you specify.
+
+#### Copy
+
+```
+bluehawk copy --destination <output-directory> <input-directory-or-file>
+```
+
+Output full bluehawk-processed input files to destination directory.
```

Comment:
Should we mention that:

- it retains the directory structure
- binary files are copied without being processed by Bluehawk
- you can use the ignore flag to add gitignore-style ignore patterns
- this respects any .gitignore files in the source directory tree. Anything in a .gitignore file is automatically used as an ignore pattern.

(Edit: Ignore stuff could go below with "ignore")

Reply:
Yes! I'll add all of this.